kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/EmbeddedKafkaIntegrationTest.java

```diff
@@ -43,91 +53,126 @@ class EmbeddedKafkaIntegrationTest {
   private KafkaProducerService<Profile> kafkaProducerService;
 
   @Autowired
-  private KafkaConsumer kafkaConsumer;
+  KafkaConsumerServiceImpl kafkaConsumerServiceImpl;
 
   @Autowired
   private ProfileRepository profileRepository;
 
   @BeforeEach
   void setup() {
-    profile = new Profile(null, TENANT_ID, PROFILE_NAME);
+    profile = new Profile(PROFILE_ID, TENANT_ID, PROFILE_NAME);
   }
 
   @Test
+  @Order(1)
   void shouldSendCreateMessageToTopicFromProducer() throws Exception {
-    profile.setId(PROFILE_ID);
     // GIVEN
     kafkaProducerService.sendMessage(PROFILE_ID.toString(), profile, KafkaAction.CREATE);
 
     // WHEN
-    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
-
+    CountDownLatch latch = new CountDownLatch(1); // block current thread execution for specific time
+    latch.await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
     // THEN
-    assertThat(messageConsumed).isTrue();
-    String expectedUpdateMessagePayload = generateExpectedPayload(version, profile, KafkaAction.CREATE);
-    assertThat(kafkaConsumer.getPayload()).isEqualTo(expectedUpdateMessagePayload);
+    assertThat(kafkaConsumerServiceImpl.getKafkaEventMessage()).isNotNull();
+
+    KafkaEventMessage expectedKafkaEventMessage = generateExpectedKafkaEventMessage(version,
+        profile,
+        KafkaAction.CREATE);
+
+    isMessageDeserialized(expectedKafkaEventMessage);
   }
 
   @Test
+  @Order(2)
   void shouldSendCreateMessageToTopicWhenProfileIsCreated() throws Exception {
     // GIVEN
     profileRepository.save(profile);
 
     // WHEN
-    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
+    CountDownLatch latch = new CountDownLatch(1); // block current thread execution for specific time
```

Comment:
Same issue here I'm concerned that there's a misunderstanding over what the latch was doing. It's effectively sleeping for the timeout

Reply:
Code changed; Now using countdown for latch as expected